### PR TITLE
WebResponseObject.Common merge partials atomic commits

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary
Merge partial classes in WebResponseObject.Common, rearrange Properties and Methods in alphabetical order and `RawContentStream` to auto-implemented property.

## PR Context
Justification: // TODO: Merge Partials 
Replaces #18692

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs [hunk 1]
```diff
@@ -62,12 +62,7 @@ public long RawContentLength
         /// <summary>
         /// Gets the response body content as a <see cref="MemoryStream"/>.
         /// </summary>
-        public MemoryStream RawContentStream
-        {
-            get { return (_rawContentStream); }
-        }
-
-        private MemoryStream _rawContentStream;
+        public MemoryStream RawContentStream { get; protected set; }
```
Comment (xtqqczze) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs:65]: Why the change to `protected`?

Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs:65]: I followed the same convention as :
```cs
public byte[] Content { get; protected set; }
```
and
```cs
public string RawContent { get; protected set; }
```

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs [hunk 2]
```diff
@@ -74,15 +74,15 @@ public long RawContentLength
         /// </summary>
         public int StatusCode
         {
-            get { return WebResponseHelper.GetStatusCode(BaseResponse); }
+            get => WebResponseHelper.GetStatusCode(BaseResponse);
```
Comment (xtqqczze) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs:77]: We could use expression-bodied members for these properties.

Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs]: Done

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs [hunk 3]
```diff
@@ -56,7 +56,7 @@ public class WebResponseObject
         /// </summary>
         public long RawContentLength
         {
-            get { return (RawContentStream is null ? -1 : RawContentStream.Length); }
+            get => (RawContentStream is null ? -1 : RawContentStream.Length);
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs]: We can remove ().

Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs]: Done

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs [hunk 4]
```diff
@@ -13,48 +13,35 @@ namespace Microsoft.PowerShell.Commands
...
         /// <summary>
-        /// Gets the response status description.
+        /// Gets the Headers property.
         /// </summary>
-        public string StatusDescription
+        public Dictionary<string, IEnumerable<string>> Headers
         {
-            get { return (WebResponseHelper.GetStatusDescription(BaseResponse)); }
-        }
+            get
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs:36]: Can convert to expression body.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs [hunk 5]
```diff
@@ -64,79 +51,32 @@ public long RawContentLength
...
 
         /// <summary>
-        /// Gets the RelationLink property.
+        /// Gets the response status description.
         /// </summary>
-        public Dictionary<string, string> RelationLink { get; internal set; }
+        public string StatusDescription => WebResponseHelper.GetStatusDescription(BaseResponse);
 
-        #endregion
+        #endregion
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs:79]: Extra end space.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs [hunk 6]
```diff
@@ -178,21 +126,26 @@ private void InitializeRawContent(HttpResponseMessage baseResponse)
             this.RawContent = raw.ToString();
         }
 
+        private static bool IsPrintable(char c)
+        {
+            return (char.IsLetterOrDigit(c) || char.IsPunctuation(c) || char.IsSeparator(c) || char.IsSymbol(c) || char.IsWhiteSpace(c));
+        }
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs:132]: Can convert to expression body.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs (hunk 5 above)
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs]: Done

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs (hunk 6 above)
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs]: Done

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs (hunk 4 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs:36]: I don't see this in new commits.

Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs:36]: I didn't do it yet, can you give me some advice on how to do it in the correct way?

Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs:36]: I added example in my previous comment.

Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs]: Thanks, done

**Review by iSazonov — COMMENTED** [review]
LGTM with two comments.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs [hunk 7]
```diff
@@ -13,48 +13,27 @@ namespace Microsoft.PowerShell.Commands
...
 
         /// <summary>
-        /// Gets the response status description.
+        /// Gets or protected sets the response body content.
         /// </summary>
-        public string StatusDescription
-        {
-            get { return (WebResponseHelper.GetStatusDescription(BaseResponse)); }
-        }
+        [SuppressMessage("Microsoft.Performance", "CA1819:PropertiesShouldNotReturnArrays")]
```
[the remark this answers is not in the document]
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs]: done

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs [hunk 8]
```diff
@@ -178,21 +118,23 @@ private void InitializeRawContent(HttpResponseMessage baseResponse)
             this.RawContent = raw.ToString();
         }
 
+        private static bool IsPrintable(char c) => char.IsLetterOrDigit(c) || char.IsPunctuation(c) || char.IsSeparator(c) || char.IsSymbol(c) || char.IsWhiteSpace(c);
```
[the remark this answers is not in the document]
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs]: done

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
@CarloToso Thanks!
There are some `this.*` prefixes which could be removed now.
